examples/vision/aodnet.py

```diff
@@ -0,0 +1,354 @@
+"""
+Title: An All-in-One Network for Dehazing and Beyond
+Author: [Soumik Rakshit](https://github.com/soumik12345)
+Date created: 2021/04/10
+Last modified: 2021/04/10
+Description: Train a lightweight Deep CNN for image denhazing.
+"""
+"""
+# Introduction
+
+Image Dehazing is an important problem is Computer Vision. It can not only be used to
+enhance photographs clicked in hazy conditions, but also can be used to augment other
+computer vision systems such as Object Detection and Seantic Segmentation as a
+pre-processing step. In this example, we implement
+[AODNet](https://arxiv.org/abs/1707.06543v1), a simple and light-weight Deep CNN model
+that directly generates a clean image from a hazy image. The end-to-end design of AODNet
```

Comment:
Suggestion:
- Clarify the abbreviation ("All-in-One Network (AOD-Net)")
- Use "AOD-Net" with a dash (as per https://arxiv.org/abs/1707.06543v1)
- Use a dash to provide a defition
- "Can generate X directly from Y" could be better
- "...deep CNN" - non capitalized
- "the AOD-Net" with the definite article ("the ... network")

```suggestion
the [All-in-One Network (AODNet)](https://arxiv.org/abs/1707.06543v1) — a simple and
lightweight deep convolutional neural network model that can generate a clean
image directly from a hazy image. The end-to-end design of the AOD-Net
```